lib/Connection.cs

```diff
@@ -36,6 +41,77 @@ public class Connection {
         }
     }
 
+    public void CreateAndSaveEncryptedKeys()
+    {
+        string publicKeyPath = $"./keys/public_key";
+        string privateKeyPath = $"./keys/private_key";
+        
+        if (KeysExists(publicKeyPath, privateKeyPath))
+        {
+            GetKeys(publicKeyPath, privateKeyPath);
+        }
+        else
+        {            
+            int keyBits = 2048;
+            var keygen = new SshKeyGenerator.SshKeyGenerator(keyBits);
+            
+            // extracting only the keys
+            string[] pubKeyList = keygen.ToRfcPublicKey().Split(" ");
+            List<String> privKeyList = new List<string>(keygen.ToPrivateKey().Split("\n"));
+
+            privKeyList.RemoveAt(privKeyList.Count - 1);
+            privKeyList.RemoveAt(privKeyList.Count - 1);
+            privKeyList.RemoveAt(0);
+
+            string priv = String.Join("\n", privKeyList);
+            string privateKey = priv.Replace("\r\n", string.Empty);
+
+            string publicKey = pubKeyList[1];
+            //
+            
+            File.WriteAllText(publicKeyPath, publicKey);
+            File.WriteAllText(privateKeyPath, privateKey);
+
+            _publicKey = publicKey;
+            _privateKey = privateKey;
+        }
+    }
+    
+    public void GetKeys(string publicKeyPath, string privateKeyPath)
+    {
+        if (File.Exists(publicKeyPath))
+        {
+            _publicKey = File.ReadAllText(publicKeyPath);
+        }
+
+        if (File.Exists(privateKeyPath))
+        {
+            _privateKey = File.ReadAllText(privateKeyPath);
+        }
+    }
+    
+    public bool KeysExists(string publicKeyPath, string privateKeyPath)
+    {
+        return File.Exists(publicKeyPath) && File.Exists(privateKeyPath);
+    }
+
+    public void GetChangedDirectory(string path)
+    {
+        if (path == "..")
+        {
+            _currentPath = Path.GetFullPath(Path.Combine(_currentPath, @"../"));
+        }
+        else if (Directory.Exists(_currentPath + path) && path != "..")
+        {
+            _currentPath = _currentPath + path;
+        }
+        else
+        {
+            Console.WriteLine("You entered the wrong path");
+        }
+        Console.WriteLine(_currentPath);
+    }
+
```

Comment:
Connection shouldn't store a path. In this PR, it should only concern itself with creating a DirectoryRequest for a specified (as in, in a paramter) path from outside.